### PR TITLE
Detail pages

### PR DESCRIPTION
**Pull request by clundin55**

### CODE REVIEW

File: yw_website/website/templates/includeable/navbar.html [hunk 1]
```diff
@@ -1,7 +1,29 @@
 {% load static %}
 <link rel="stylesheet" type="text/css" href={% static "design/nav_bar.css" %}>
+<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
+<link rel="stylesheet" href="https://www.w3schools.com/lib/w3-theme-black.css">
+<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto">
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+{% comment %}
+<!-- <div class="w3-top">
```
Comment (aniehuser) [inline yw_website/website/templates/includeable/navbar.html]: what is this?

File: yw_website/website/templates/pages/detailed_workflow.html [hunk 2]
```diff
@@ -1,88 +1,5 @@
-{% load staticfiles %}
```
Comment (aniehuser) [inline yw_website/website/templates/pages/detailed_workflow.html:1]: i think the inheritance is a little wierd here. how does this reduce redundancy?

More specifically, what does `_base_detailed_workflow.html` have in common with other pages we might have?

File: yw_website/website/views.py [hunk 3]
```diff
@@ -1,79 +1,73 @@
...
-    documents_list = Workflow.objects.all().filter(user=request.user)
-    for document in documents_list:
-        latest_version = Version.objects.filter(workflow_id=document.id).order_by('last_modified').first()
-        document.graph = latest_version.yw_graph_output if latest_version is not None else ""
-    
-    paginator = Paginator(documents_list, 10)
+    workflow_list = Workflow.objects.all().filter(user=request.user).exclude(version__isnull=True)
+    for workflow in workflow_list:
+        latest_version = Version.objects.filter(workflow=workflow).order_by('last_modified').first()
+        if latest_version is None:
```
Comment (aniehuser) [inline yw_website/website/views.py:38]: will we ever have a case where a workflow doesn't have a version?

File: yw_website/website/views.py [hunk 4]
```diff
@@ -1,79 +1,73 @@
...
+    
+    except ObjectDoesNotExist:
       return redirect(home)
 
-def run_detail(request):
-    document = Run.objects.get(id="1")
+def run_detail(request, run_id):
+    try:
+        document = Run.objects.get(pk=run_id)
+    except Run.DoesNotExist:
```
Comment (aniehuser) [inline yw_website/website/views.py:71]: we probably want to throw a 404 error here

File: yw_website/yw_website/urls.py [hunk 5]
```diff
@@ -44,8 +44,8 @@
     ###
     path('', yw_website_views.home, name='home'),
     path('my_workflows/', yw_website_views.my_workflows, name='my_workflows'),
-    path('detailed_workflow/<int:document_id>/', yw_website_views.detailed_workflow, name='detailed_workflow'),
-    path('run_detail/', yw_website_views.run_detail, name='run_detail'),
+    path('detailed_workflow/<int:workflow_id>/<int:version_id>/', yw_website_views.detailed_workflow, name='detailed_workflow'),
```
Comment (aniehuser) [inline yw_website/yw_website/urls.py]: a better way to format this route would be `detailed_workflow/<int:workflow_id>/version/<int:version_id>/`

File: yw_website/yw_website/settings.py [hunk 6]
```diff
@@ -77,13 +81,19 @@
...
             ],
         },
     },
 ]
 
 WSGI_APPLICATION = 'yw_website.wsgi.application'
 
+AUTHENTICATION_BACKENDS = (
+    'django.contrib.auth.backends.ModelBackend',
+    'allauth.account.auth_backends.AuthenticationBackend',
```
Comment (aniehuser) [inline yw_website/yw_website/settings.py:95]: would you be willing to drop some links in the description of this merge request so we can all reference the authentication framework you used later?

File: yw_website/website/views.py [hunk 7]
```diff
@@ -1,79 +1,73 @@
...
+
+            runs = Run.objects.filter(version=version)
+            info = {'workflow': workflow, 'version': version, 'versions':versions ,'run_list':runs, 'form': form}
             return render(request, 'pages/detailed_workflow.html', info)
-    except Workflow.DoesNotExist:
+        elif request.method == "POST":
+            new_version = request.POST['version_id']
+            return redirect('/detailed_workflow/{}/{}/'.format(workflow_id, new_version))
+    
+    except ObjectDoesNotExist:
```
Comment (aniehuser) [inline yw_website/website/views.py:65]: again, we should return a 404 instead of redirecting

File: yw_website/website/views.py [hunk 8]
```diff
@@ -1,79 +1,73 @@
...
+            version = Version.objects.get(pk=version_id)
+            versions = Version.objects.filter(workflow=workflow)
+
+            runs = Run.objects.filter(version=version)
+            info = {'workflow': workflow, 'version': version, 'versions':versions ,'run_list':runs, 'form': form}
             return render(request, 'pages/detailed_workflow.html', info)
-    except Workflow.DoesNotExist:
+        elif request.method == "POST":
+            new_version = request.POST['version_id']
+            return redirect('/detailed_workflow/{}/{}/'.format(workflow_id, new_version))
```
Comment (aniehuser) [inline yw_website/website/views.py]: I like the way some fo these endpoints are formatted for reference: https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9

File: yw_website/website/templates/account/base.html [hunk 9]
```diff
@@ -0,0 +1,49 @@
+
```
Comment (aniehuser) [inline yw_website/website/templates/account/base.html:1]: what is this template for? are we straying from the `extendable`/`includeable`/`pages` file structure format?

Should we be inheriting from `__base.html` ?

**Comment by aniehuser** [comment]
We should really be putting one task in a merge request, it makes it pretty hard to review.

File: yw_website/website/templates/includeable/navbar.html (hunk 1 above)
Comment (aniehuser) [reply yw_website/website/templates/includeable/navbar.html]: lets remove this as from discussion

File: yw_website/website/views.py (hunk 3 above)
Comment (aniehuser) [reply yw_website/website/views.py:38]: add TODO for db population scripts

File: yw_website/yw_website/settings.py (hunk 6 above)
Comment (clundin55) [reply yw_website/yw_website/settings.py:95]: https://django-allauth.readthedocs.io/en/latest/

**Review by aniehuser — APPROVED** [review]
sweet

**Pull request merged**